### PR TITLE
Hermetic build failure in konflux for odh-mm-rest-proxy component

### PR DESCRIPTION
**Pull request by spolti**
## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -128,7 +134,7 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
     GOARCH=${TARGETARCH:-amd64} \
     CGO_ENABLED=0 \
     GO111MODULE=on \
-    go mod tidy && go build -a -o /go/bin/server ./proxy/
+    go build -a -o /go/bin/server ./proxy/
```
Comment (Jooho) [inline Dockerfile:137]: `go mod tidy` impacts something?

Comment (spolti) [reply Dockerfile:137]: yes, as it tries to clean unneeded dependencies it leads to issues like:

```
go: github.com/kserve/rest-proxy/proxy imports
	sigs.k8s.io/controller-runtime/pkg/log/zap tested by
	sigs.k8s.io/controller-runtime/pkg/log/zap.test imports
	github.com/onsi/ginkgo/v2: github.com/onsi/ginkgo/v2@v2.9.1: reading file:///cachi2/output/deps/gomod/pkg/mod/cache/download/github.com/onsi/ginkgo/v2/@v/v2.9.1.zip: no such file or directory
```

Talking with a colleague from Konflux team, at least for builds there, `go mod tidy` should not be executed.

Comment (Jooho) [reply Dockerfile:137]: So this issue only happens in konflux? I prefer to add this command specifically for cleaning and optimizing the go.mod and go.sum files. 

The error message shows that ginkgo was not found. I think we should check why it is not there. WDYT?

Comment (Jooho) [reply Dockerfile:137]: yes we need the process to execute `go mod tidy` before sending PR. 

ok I am ok with this.

**Review by israel-hdez — APPROVED** [review]
/lgtm

**Pull request merged**